datafusion/core/src/execution/context.rs

```diff
@@ -518,7 +518,7 @@ impl SessionContext {
                 let physical = DataFrame::new(self.state(), input);
 
                 let batches: Vec<_> = physical.collect_partitioned().await?;
-                let table = Arc::new(MemTable::try_new(schema, batches)?);
+                let table = Arc::new(MemTable::new_not_registered(schema, batches));
```

Comment:
It seems like the core problem here is `schema` (aka the schema of `input`) does not actually match the schema of the batches that are produced. 

As I think @comphead is suggesting in https://github.com/apache/arrow-datafusion/pull/6566#pullrequestreview-1465439929, this PR seems to be trying to workaround the deeper problem of the window exec producing an output schema (different column names) than the LogicalPlan says. 

Have you looked into making the names consistent?

Reply:
Yes, my first suggestion is to use longer version of the name in planner (what is done in the LogicalPlan). However, there need to be many changes in tests, and the BoundedWindowAggExec lines may become too long. If it is not a problem, we can solve the issue like [that](https://github.com/apache/arrow-datafusion/pull/6566#issuecomment-1580178810)

Reply:
I think using the same names in physical and logical plans is preferable because the rest of the parts of the code expects this and sometimes makes assumptions that it is the case (because it mostly is). 

If we don't make the logical and physical plans match up, I predict we will continue to hit a long tail of bugs related to schema mismatches, only when using window functions related to the discrepancy.

If the long display name is a problem (and I can see how it would be) perhaps we can figure out how to make `display_name` produce something shorter for window functions other than serializing the entire window definition

Here is what postgres does:

```sql
postgres=# select first_value(x) over (order by x) from foo;
 first_value
-------------
           1
(1 row)
```

We probably need to do something more sophisticated as DataFusion needs distinct column names.